src/ogt_vox.h

```diff
@@ -797,6 +804,17 @@
         size_t count;      // size of the array
     };
 
+    // progress callback function.
+    static ogt_vox_progress_callback_func g_progress_callback_func = NULL;
+    static void* g_progress_callback_user_data = NULL;
```

Comment:
Not sure if adding more global vars is a good thing here. I would vote for another version besides the existing load/save function that take a context as variable. That way we wouldn't break the existing api - but would still allow to use the library in different context with different settings.

Even though I would say that this api breakage would be easy to fix for users. `ogt_create_context()`, `ogt_delete_context()`, and `ogt_vox_read_scene(ctx, buffer, buffer_size)`

as far as I could see only `g_alloc_func` and `g_free_func` are static variables - and even those should go into a context variable to be able to use the library with different settings or allocators or loading multiple vox models with different progress indicators for different widgets - who knows.

Reply:
A context is a good idea.

If you don't want to break the API there could be a global default context, and the current interface would use this.

```C
static ogt_context g_default_ctx;
ogt_vox_read_scene(buffer, buffer_size)
{
    ogt_vox_read_scene(g_default_ctx, buffer, buffer_size);
}
```

Along with `ogt_get_default_context()` and `ogt_set_default_context()` functions.

Thoughts?

Reply:
Though there is precedent for global state (the memory allocator functions), it's not great with multiple threads unless you use a thread-safe allocator. 

On the progress callback side, it's a little less clear how to do that multithreaded without a per-thread context, so I find myself agreeing with ideas from both @mgerhardy and @dougbinks here.

At the minumum we should have an additional flavor of ogt_vox_read_scene that takes a context. eg.
```
ogt_vox_scene* ogt_vox_read_scene( ogt_vox_context* ctx, const uint8_t* buffer, uint32_t buffer_size );
ogt_vox_scene* ogt_vox_read_scene_with_flags( ogt_vox_context* ctx, const uint8_t* buffer, uint32_t buffer_size, uint32_t flags );
```

... whereas the existing ones can simply call into these with a NULL context:

```
ogt_vox_scene* ogt_vox_read_scene( buffer, buffer_size ) {
  return ogt_vox_read_scene( NULL, buffer, buffer_size );
}
ogt_voxe_scene* ogt_vox_read_scene_with_flags( &ctx, buffer, buffer_size, flags ) {
  return ogt_vox_read_scene_with_flags( NULL, buffer, buffer_size, flags );
}
```

Of course, I think a global context is probably okay in the short-term, but I'd like to steer away from having the library itself retain that global state (and eventually for the memory allocators functions too)